client/player.js

```diff
@@ -76,6 +88,8 @@ var Player = function (position_, world_) {
 
   this.setLoadout = function(l){
     loadout = l;
+    loadout.magazine = shotsLeft[loadout.name]
```

Comment:
the number of shots left is already stored in loadout.magazine

Reply:
Each time the loadout is changed, a new instance of the loadout is created, which re-sets the shots left count. I think we either need to store an instance of each loadout type for each client, or store the shots left per client.